src/wp-admin/site-editor.php

```diff
@@ -19,7 +19,7 @@
 	);
 }
 
-if ( ! wp_is_block_theme() ) {
+if ( ! ( current_theme_supports( 'block-template-parts' ) || wp_is_block_theme() ) ) {
 	wp_die( __( 'The theme you are currently using is not compatible with Full Site Editing.' ) );
 }
 
```

Comment:
@hellofromtonya I belive something like this would solve the case you uncovered in your testing here: https://github.com/WordPress/wordpress-develop/pull/3234#issuecomment-1247048286

> With the block-fragments (classic theme with block-based templates) enabled, I am able to directly access /wp-admin/site-editor.php but it's a white screen. No errors in the browser's console or error logs.

```suggestion
$is_template_part_editor = isset( $_GET['?postType'] ) && 'wp_template_part' === $_GET['?postType'];
if ( ! wp_is_block_template_theme() && ! $is_template_part_editor ) {
	wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
}
```

As far as I'm aware that was not a known issue and no patch for it had been created. 

@Mamaduka asked me to keep an eye on this PR since he is out traveling. I will create a PR on the Gutenberg Repo to fix the issue there aswell.

Reply:
Actually in looking at the code in Gutenberg it was handled as a completely separate admin page. So this issue only exists in this PR and only needs to get fixed here. No backport to Gutenberg is needed. 

https://github.com/WordPress/gutenberg/blob/d19258cb5b85cea52e85a3e86c9550dd48d44fca/lib/compat/wordpress-6.1/template-parts-screen.php#L5

Reply:
In order to make it easier to test, I have also created a separate PR against @Mamaduka's branch which applies the same fix as the one I have shown in the suggestion above: https://github.com/Mamaduka/wordpress-develop/pull/48

Reply:
Yes, that does resolve it. Well done @fabiankaegy - thank you!

Reply:
Thank you!